source/shadow/SampleShadowFeature.h

```diff
@@ -0,0 +1,192 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef AWS_IOT_DEVICE_CLIENT_SAMPLESHADOW_H
+#    define AWS_IOT_DEVICE_CLIENT_SAMPLESHADOW_H
+
+#endif // AWS_IOT_DEVICE_CLIENT_SAMPLESHADOW_H
+
+#include "../ClientBaseNotifier.h"
+#include "../Feature.h"
+#include "../SharedCrtResourceManager.h"
+#include "../config/Config.h"
+#include "../util/FileUtils.h"
+#include <aws/iotshadow/IotShadowClient.h>
+
+namespace Aws
+{
+    namespace Iot
+    {
+        namespace DeviceClient
+        {
+            namespace Shadow
+            {
+                class SampleShadowFeature : public Feature
+                {
+                  public:
+                    int init(
+                        std::shared_ptr<SharedCrtResourceManager> manager,
+                        std::shared_ptr<ClientBaseNotifier> notifier,
+                        const PlainConfig &config);
+
+                    // Interface methods defined in Feature.h
+                    std::string getName() override;
+
+                    int start() override;
+
+                    int stop() override;
+
+                  private:
+                    /**
+                     * \brief the ThingName to use
+                     */
+                    std::string thingName;
+                    static constexpr char TAG[] = "SampleShadowFeature.cpp";
+                    /**
+                     * \brief The resource manager used to manage CRT resources
+                     */
+                    std::shared_ptr<SharedCrtResourceManager> resourceManager;
+                    /**
+                     * \brief An interface used to notify the Client base if there is an event that requires its
+                     * attention
+                     */
+                    std::shared_ptr<ClientBaseNotifier> baseNotifier;
+                    /**
+                     * \brief Whether the DeviceClient base has requested this feature to stop
+                     */
+                    std::atomic<bool> needStop{false};
+                    /**
+                     * \brief Name of shadow
+                     */
+                    std::string shadowName;
+                    /**
+                     * \brief Location of file containing data to upload into shaodow
+                     */
+                    std::string inputFile;
+                    /**
+                     * \brief Location of file to write the latest shadow document to
```

Comment:
What will be the default file location for these input/output files? We should have default file location in case user does not provide any.

Reply:
see my above reply on default input/ouput